Library/Homebrew/cask/cask.rb

```diff
@@ -283,7 +283,10 @@ def populate_from_api!(json_cask)
       @tap_git_head = json_cask.fetch(:tap_git_head, "HEAD")
 
       @ruby_source_path = json_cask[:ruby_source_path]
-      @ruby_source_checksum = json_cask[:ruby_source_checksum].freeze
+
+      ruby_source_sha256 = json_cask.dig(:ruby_source_checksum, :sha256)
+      ruby_source_sha256 ||= json_cask[:ruby_source_sha256]
```

Comment:
What do you think about either `# TODO`ing or splitting into separate functions these paths so they don't outlast the move to non-internal JSON?

Reply:
Good idea. It is quite ugly right now.